test/new-e2e/tests/sysprobe-functional/usmtest/test_tags.ps1

```diff
@@ -0,0 +1,130 @@
+
+
+param(
+    [Parameter(Mandatory=$true)][string]$TargetHost,
+    [Parameter(Mandatory=$true)][string]$TargetPort,
+    [Parameter(Mandatory=$true)][string]$TargetPath,
+    [Parameter(Mandatory=$true)][string[]]$ExpectedClientTags,
+    [Parameter(Mandatory=$true)][string[]]$ExpectedServerTags,
+    [Parameter(Mandatory=$true)][string]$ConnExe
+)
+function stop-servicesfortest {
+    # disable process agent so that it doesn't query the connections endpoint while we're running
+    stop-service -force datadog-process-agent -ErrorAction Stop
+
+    # stop system probe to clean out any connections we're not interested in
+    stop-service -force datadog-system-probe -ErrorAction Stop
+}
+
+function make-connectionrequest {
+    param(
+        [Parameter(Mandatory=$true)] [string]$ConnExe,
+        [Parameter(Mandatory=$true)] [string]$TargetHost,
+        [Parameter(Mandatory=$false)] [string]$Port="80",
+        [Parameter(Mandatory=$false)] [string]$TargetPath="/"
+    )
+    $list = @($TargetHost, $Port, $TargetPath)
+
+    $targetpid = (start-process $ConnExe -passthru -NoNewWindow -ArgumentList $list).id    
+    return $targetpid
+}
+
+function get-connectionsendpoint {
+    $payload = ((iwr -UseBasicParsing -DisableKeepAlive http://localhost:3333/network_tracer/connections).content | ConvertFrom-Json)
```

Comment:
This will break once the named pipe is in place.

Reply:
Yes.  This is well understood and you and I discussed this on slack.
However, at this moment, this is the mechanism for retrieving the connections.